airflow/models.py

```diff
@@ -726,6 +731,7 @@ def __init__(self, task, execution_date, state=None):
         self.pool = task.pool
         self.priority_weight = task.priority_weight_total
         self.try_number = 0
+        self.job_id = job_id
```

Comment:
@mistercrunch what does `TaskInstance.job_id` do? it seems to be used in a few places but I can't find any place that sets it... is it a foreign key of some kind?